### DISCUSSION TITLE
switchNetwork function possibly undefined ?

### DISCUSSION THREAD

**Discussion by Xeway** [wevm/wagmi#343] [Q&A] [answered]
Gm,
I've two buttons that let the user switch network to Kovan (1st button) or to mainnet (2nd button) in case of the user go into a wrong network (only Kovan and mainnet allowed in my dapp).
But typescript let me know a problem : `Impossible to call an object that may have an 'undefined' value` (sorry maybe it's not the exact message bc my IDE is in french but you get the idea).
And so if I click on the button it also throws an error that says : `TypeError: switchNetwork is not a function`.
Here are the two buttons : 
```
<button className="switch-network-btn"
onClick={() => switchNetwork(42).then(() => setNetworkError(false))}>
Switch to Kovan
</button>

<button className="switch-network-btn"
onClick={() => switchNetwork(1).then(() => setNetworkError(false))}>
Switch to Ethereum
</button>
```

Can some help me please ? Thanks in advance !

**Comment by tmm** [comment]
What wallet are you using? Some wallets do not support programmatic switching so the function is not defined in that case.

**Comment by Xeway** [comment]
I use Metamask so it should work, right ?
And if some wallets don't support switching, is there a way to prevent that error ? (something like if wallet === "Metamask" show that button else don't show that button)

**Reply by tmm** [reply]
MM should work, correct. Can you put together an example (CodeSandbox, GH repo, etc)?

You can show/hide the buttons if `switchNetwork` is defined/undefined.

**Comment by Xeway** [comment]
[Sure here's a CodeSandbox](https://codesandbox.io/s/hungry-resonance-1l84ro?file=/components/ConnectionButtons.tsx)
It's not working because I don't have add all the files + I don't have my API keys but you can check the most important files : ConnectionButtons.tsx (and maybe NetworkInfo.tsx).
Don't hesitate if you want more information and thank you for your help !

**Comment by Xeway** [comment, marked as answer]
Okay so I fixed the problem, I just have to embed my buttons with {switchNetwork && /*button*/}

**Comment by theshadowagent** [comment]
This error also appears if you have both Metamask & Rabby Wallet extensions installed, and you connected Metamask wallet (`wagmi` & `connectkit` expect you to connect via Rabby Wallet). Solved by disabling Rabby.
